src/main/kotlin/org/rust/cargo/runconfig/RsExecutableRunner.kt

```diff
@@ -74,9 +78,20 @@ abstract class RsExecutableRunner(
 
         val runCargoCommand = state.prepareCommandLine()
         val (_, executableArguments) = parseArgs(runCargoCommand.command, runCargoCommand.additionalArguments)
+
+        val workingDirectory = if (runCargoCommand.command == "test") {
+            val srcPath = artifact?.target?.src_path
+            val file = srcPath?.let { LocalFileSystem.getInstance().refreshAndFindFileByPath(it) }
```

Comment:
Let's avoid using VFS here because it has some contract how it should be called not to get dead lock. As I can see, artifact json object has `package_id` field (at least at 1.48 and above, i.e. when build view works in the plugin) so we can use this id to find the corresponding package and avoid using VFS at all

Reply:
Do you mean something like this?

```rust
cargoProjects.allProjects.mapNotNull { it.workspace?.findPackage(packageId) }.singleOrNull()
```

Reply:
Yep